components/elm/docs/dev-guide/testing.md

```diff
@@ -0,0 +1,183 @@
+
+```
+git checkout cfb7fc2b2
+Note: checking out 'cfb7fc2b2'.
+
+You are in 'detached HEAD' state. You can look around, make experimental
+changes and commit them, and you can discard any commits you make in this
+state without impacting any branches by performing another checkout.
+
+If you want to create a new branch to retain commits you create, you may
+do so (now or later) by using -b with the checkout command again. Example:
+
+  git checkout -b <new-branch-name>
+
+HEAD is now at cfb7fc2b2... Merge branch 'qzhu-lbl/lnd/ch4_inundation_bugfix' (PR #2814)
+
+git submodule update --init
+```
+
+Now we will use `cime/scripts/create_test` to generate baseline for the `e3sm_land_developer`
+test suite.
+
+```
+cd cime/scripts
+```
+
+A few things that you need to decide at this stage include the directory location where the baselines will be saved, the name and ID for the baselines, the project allocation that you will use for running the simulation, if you would like to recieve email notirications about tests, etc.
+
+```
+# Define the directory to hold the baseline 
+export MY_BASELINE_DIR=/global/cscratch1/sd/gbisht/e3sm_baselines
+
+# Do you want to use a name for the baseline? 
+# One choice could be the git hash that is being used to generate the baselines.
+export BASELINE_NAME=cfb7fc2b2
+
+# Let's set TEST_ID to be same as BASELINE_NAME
+export TEST_ID=${BASELINE_NAME}
+
+# If you are a member of E3SM, you could use 'e3sm' project allocation
+export PROJECT=e3sm
+
+# Set your email
+export MAIL_USER=<your-mail@something>
+
+```
+
+Use `./create_test --help` to get a complete list of arguments. Below are some additional
+useful arguments for `./create_test`:
+
+```
+# Other arguments
+# -v         : Verbose option
+# -g         : Generate the baseline
+# -q         : If you want to use a particular job queue (e.g. 'debug' queue on NERSC)
+# --walltime : Specify the wall time for jobs (e.g. 30 min is max allowable for 'debug' queue on NERSC)
+# --mail-user: If you want to receive emails about your jobs
+# --mail-type: When to receive emails. Options are: never, all, begin, end, fail.
+# -j         : Number of parallel jobs
+```
+
+Now run the `e3sm_land_developer`
+
+```
+./create_test e3sm_land_developer  \
+--baseline-root ${MY_BASELINE_DIR} \
+-b ${BASELINE_NAME}                \
+-t ${TEST_ID}                      \
+-q regular                         \
+-p ${PROJECT}                      \
+--walltime 00:30:00                \
+--mail-user $MAIL_USER             \
+--mail-type all                    \
+-g                                 \
+-v                                 \
+-j 4
+```
+
+The cases would be named `*.G.*` to denote one is generating the baselines.
+It will take a while to compile all the cases and submit the code.
```

Comment:
Would it be helpful to suggest using a `screen -S tests` environment so the tests don't get interrupted during compilation?

Reply:
Good suggestion.